zerver/actions/create_user.py

```diff
@@ -182,9 +182,12 @@ def process_new_human_user(
         streams = []
         acting_user = None
 
-    # If the user's invitation didn't explicitly list some streams, we
-    # add the default streams
-    if len(streams) == 0:
+    user_was_invited = prereg_user is not None and (
+        prereg_user.referred_by is not None or prereg_user.multiuse_invite is not None
+    )
+    # If the Preregistration object didn't explicitly list some streams (it happens when user
+    # directly signs up without any invitation), we add the default streams
+    if len(streams) == 0 and not user_was_invited:
         streams = get_default_subs(user_profile)
 
     for default_stream_group in default_stream_groups:
```

Comment:
I am not sure what we want do for streams in `default_stream_groups`. We previously added the streams in `default_stream_groups` irrespective of the streams passed during invite.